src/pages/Register/Register.tsx

```diff
@@ -42,7 +42,11 @@ const Register = ({ setUser }: RegisterProps) => {
       navigate("/");
     } catch (error: unknown) {
       if (error instanceof FirebaseError) {
-        if (error.code === "auth/email-already-in-use") setIsEmailExist(true);
+        if (error.code === "auth/email-already-in-use")
+          setFirebaseError("emailError");
```

Comment:
I think you can just add the text you want to display to state.

```jsx
if (error.code === "auth/email-already-in-use")
          setFirebaseError("Email Already Exists");
 }         
 
 // Then in return, you only need one check
 
{firebaseError && <p className="register-page__email-exists">
{firebaseError }</p>}
```

Reply:
Done!